### PR TITLE
Fixed visibility property for visuals with shadow

### PR DESCRIPTION
**Pull request by aborziak-ms**
Fixing translation of shadow effect. If source of the shadow is not visible it does not make shadow disappear, so here I'm pushing `IsVisible` property up so that this property controls visibility of both, source and shadow visuals.

### CODE REVIEW

File: source/LottieToWinComp/Effects.cs [hunk 1]
```diff
@@ -267,6 +272,37 @@ void EmitIssueAboutUnsupportedEffect(Effect? effect, string effectName)
...
+                {
+                    Animate.WithKeyFrame(context, target, animator.AnimatedProperty, (KeyFrameAnimation_)animator.Animation);
+                }
+                else if (animator.Animation is ExpressionAnimation)
+                {
+                    Animate.WithExpression(target, (ExpressionAnimation)animator.Animation, animator.AnimatedProperty);
+                }
+                else
+                {
+                    throw new ArgumentException("IsVisible animation should be keyframe of expression");
```
Comment (raymond-wh-leung) [inline source/LottieToWinComp/Effects.cs:295]: >of  [](http://example.com/codeflow?start=88&length=3)

nit: or?

File: source/LottieToWinComp/Effects.cs [hunk 2]
```diff
@@ -267,6 +272,37 @@ void EmitIssueAboutUnsupportedEffect(Effect? effect, string effectName)
...
+                }
+
+                source.StopAnimation(animator.AnimatedProperty);
+            }
+
+            // Here we are moving default value of "IsVisible" to target
+            target.IsVisible = source.IsVisible;
```
Comment (raymond-wh-leung) [inline source/LottieToWinComp/Effects.cs:302]: > target.IsVisible = source.IsVisible; [](http://example.com/codeflow?start=11&length=37)

If the Animate.WithKeyFrame() or WithExpression() calls StartAnimation() that targets IsVisibile, this set here (if the the value gets changed), will stop the animation above, maybe this needs to be set first before the Animate.WithKeyFrame() calls above?

File: source/LottieToWinComp/Effects.cs (hunk 1 above)
Comment (aborziak-ms) [reply source/LottieToWinComp/Effects.cs]: fixed

File: source/LottieToWinComp/Effects.cs [hunk 3]
```diff
@@ -242,6 +242,11 @@ void EmitIssueAboutUnsupportedEffect(Effect? effect, string effectName)
             result.Size = size;
             result.Children.Add(blurResult);
 
+            // We need to transfer "IsVisible" property from source to result
+            // because result is now the root visual for both shadow and source.
+            // Otherwise "IsVisible" property will be applied only to source.
+            MoveIsVisibleProperty(context, source, result);
```
Comment (raymond-wh-leung) [inline source/LottieToWinComp/Effects.cs:248]: >MoveIsVisibleProperty [](http://example.com/codeflow?start=12&length=21)

nit: what do you think about TransferIsVisible or PropagateIsVisible?

File: source/LottieToWinComp/Effects.cs (hunk 2 above)
Comment (aborziak-ms) [reply source/LottieToWinComp/Effects.cs]: This class is just a "mock class" of a real WinComp class (Lottie-Windows\source\WinCompData\Visual.cs) so it just stores the data in it, after all the translation is done this class is used to instantiate real WinComp instances. So it should be fine, but I reordered it anyway

File: source/LottieToWinComp/Effects.cs (hunk 3 above)
Comment (aborziak-ms) [reply source/LottieToWinComp/Effects.cs]: Transfer sounds good, done

File: source/LottieToWinComp/Effects.cs (hunk 2 above)
Comment (raymond-wh-leung) [reply source/LottieToWinComp/Effects.cs]: Gotcha.

**Review by raymond-wh-leung — APPROVED** [review]
:shipit:

**Pull request merged**